### DISCUSSION TITLE
can't get leds to work on STM32F411

### DISCUSSION THREAD

**Discussion by gitrimius** [shanteacontrols/OpenDeck#73] [Q&A]
Hi,
I am trying to get some leds going.
My test-setup is the following:

- using the default STM32F411 .yml from the TARGET dir. I build the fw. It seems everything works fine except for the leds.
- 1 button on B2 to GND. 
- 1 led on A15 to Resistor 220 to GND
- nothing else is connected to the board.

By pressing the button (internal note etc...) the led is not lighting.
When I measure between A15 and GND when I press the button the voltage goes from 0 to around 0,15v.
I noticed that the settings for flashing, startup animation etc are not saved when I unplug the board...
Factory default reset did not help.

Any idea what I might be doing wrong?
tnx!

**Comment by paradajz** [comment]
Next time you start a discussion, write me the exact release you're using or if you're building from master. The LED outputs are now fixed (pushed to master). As for your configuration, I can't reproduce that issue - the settings are retained after I plug and unplug the board.

**Reply by gitrimius** [reply]
Hi,
Sorry for my beginners mistakes. This is all new to me...

fyi. I was building based on 5.4.5. After recloning the repository and rebuilding it works like a charm, great!

Just a thought:
I now have 8 switches (no buttons) that are linked (via note) to the 8 leds. When I switch Presets the state of the switches is not represented by the leds... (it is when powering on due to the animation). 
It would be great to allow for 'animation when switching preset', or automatically burst the 'button' states.. That way the leds would represent the state of the switches.

anyway, great work!

**Reply by paradajz** [reply]
I'm not quite sure I understand. Can you make a short video with the explanation of the problem so that I can visualize it?

**Reply by gitrimius** [reply]
sure.
enable the subs for a short description. Around the middle of the movie I switch from Preset 1 to Preset 2. The settings for leds and buttons are the same on both Presets. Notice how the leds go off when switching the Presets although the switches stay on.

https://youtu.be/1M-xNZYa_O4

**Reply by paradajz** [reply]
Thanks. So basically the same thing as forced resend of analog values, just for buttons. It's pushed on master but I can't test it right now. Try it out and let me know.

**Reply by gitrimius** [reply]
just tested. Works great. Thanks a lot!

**Comment by paradajz** [comment]
Part of new firmware now:
https://github.com/paradajz/OpenDeck/releases/tag/v5.5.0
